docs/OV_Runtime_UG/ShapeInference.md

```diff
@@ -122,104 +116,126 @@ The Inference Engine provides a special mechanism that allows adding support of
 
-You can change input shapes multiple times using the `IENetwork.reshape` and `IENetwork.batch_size` methods in any order. If a model has a hard-coded batch dimension, use `IENetwork.batch_size` first to change the batch, then call `IENetwork.reshape` to update other dimensions, if needed.
+Once the input shape of [Model](api/ie_python_api/_autosummary/openvino.runtime.Model.html) is set, call the [compile_model](api/ie_python_api/_autosummary/openvino.runtime.compile_model.html) method to get a [CompiledModel](api/ie_python_api/_autosummary/openvino.runtime.CompiledModel.html) object for inference with updated shapes.
 
-Inference Engine takes three kinds of a model description as an input, which are converted into an IENetwork object:
+There are other approaches to change model input shapes during the stage of <a href="_docs_MO_DG_prepare_model_convert_model_Converting_Model.html#when-to-specify-input-shapes">IR generation</a> or [Model creation](../OV_Runtime_UG/model_representation.md).
 
-1. Intermediate Representation (IR) through `IECore.read_network`
-2. ONNX model through `IECore.read_network`
-3. nGraph function through the constructor of IENetwork
+### Dynamic Shape Notice
 
-IENetwork keeps an `ngraph::Function` object with the model description internally. The object should have fully defined input shapes to be successfully loaded to the Inference Engine plugins. To resolve undefined input dimensions of a model, call the `IENetwork.reshape` method providing new input shapes before loading to the Inference Engine plugin.
+Shape-changing functionality could be used to turn dynamic model input into a static one and vice versa.
+To learn more about dynamic shapes in OpenVINO please see a <a href="_docs_OV_Runtime_UG_ov_dynamic_shapes.html">dedicated article</a>.
 
-Run the following code right after IENetwork creation to explicitly check for model input names and shapes:
+### Usage of Reshape Method <a name="usage_of_reshape_method"></a>
 
-To feed input data of a shape that is different from the model input shape, reshape the model first.
+The primary method of the feature is [Model.reshape](api/ie_python_api/_autosummary/openvino.runtime.Model.html#openvino.runtime.Model.reshape). It is overloaded to better serve two main use cases:
 
-Once the input shape of IENetwork is set, call the `IECore.load_network` method to get an ExecutableNetwork object for inference with updated shapes.
+1) To change input shape of model with single input you may pass new shape into the method. Please see the example of adjusting spatial dimensions to the input image:
```

Comment:
Looks like here we have the same text like for C++ part.
@ilya-lavrenov @andrew-zaytsev @vyashina It returns us to our discussion about C++/Python tabs for snippets or for a page.
I think it can be aligned separately.